eventuate-core/src/main/scala/com/rbmhtechnology/eventuate/AbstractEventsourcedView.scala

```diff
@@ -68,109 +125,127 @@ private[eventuate] abstract class AbstractEventsourcedComponent extends Eventsou
    * Returns a receive object that defines the actor's command handling behaviour.
    * Use [[$akkaReceiveBuilder ReceiveBuilder]] to define the behaviour.
    *
-   * Takes precedence over [[setOnCommand]].
-   *
    * @see [[EventsourcedView]]
    */
-  def createOnCommand(): AbstractActor.Receive = commandBehaviour.getOrElse(AbstractActor.emptyBehavior)
+  def createOnCommand(): AbstractActor.Receive = AbstractActor.emptyBehavior
 
-  final override def onCommand =
-    createOnCommand().onMessage.asInstanceOf[Receive]
+  final override def onCommand = createOnCommand().asScala
 
   /**
    * Java API of the [[EventsourcedView.onSnapshot snapshot]] handler.
    *
    * Returns a receive object that defines the actor's snapshot handling behaviour.
    * Use [[$akkaReceiveBuilder ReceiveBuilder]] to define the behaviour.
    *
-   * Takes precedence over [[setOnSnapshot]].
-   *
    * @see [[EventsourcedView]]
    */
-  def createOnSnapshot(): AbstractActor.Receive = snapshotBehaviour.getOrElse(AbstractActor.emptyBehavior)
+  def createOnSnapshot(): AbstractActor.Receive = AbstractActor.emptyBehavior
 
-  final override def onSnapshot =
-    createOnSnapshot().onMessage.asInstanceOf[Receive]
+  final override def onSnapshot = createOnSnapshot().asScala
 
   /**
    * Java API of the [[EventsourcedView.onRecovery recovery]] handler.
    *
    * Returns a result handler that defines the actor's recover handling behaviour.
    * Use [[ResultHandler]] to define the behaviour.
    *
-   * Takes precedence over [[setOnRecover]].
-   *
    * @see [[EventsourcedView]]
    */
-  def onRecover: ResultHandler[Unit] = recoveryHandler.getOrElse(ResultHandler.none[Unit])
+  def createOnRecovery(): ResultHandler[Unit] = ResultHandler.none[Unit]
+
+  override final def onRecovery: Handler[Unit] = createOnRecovery().asScala
 
   /**
-   * Java API that sets this actor's [[EventsourcedView.onCommand command]] handler.
-   *
-   * Supplied with a receive object that defines the actor's command handling behaviour.
-   * Use [[$akkaReceiveBuilder ReceiveBuilder]] to define the behaviour.
-   *
-   * If [[onCommand]] is implemented, the supplied behaviour is ignored.
+   * Creates a new empty `ReceiveBuilder`. Mimics [[AbstractActor.receiveBuilder()]].
+   */
+  final def receiveBuilder(): ReceiveBuilder = ReceiveBuilder.create()
+
+  /**
+   * Returns this AbstractActor's ActorContext. Mimics [[AbstractActor.getContext()]].
+   */
+  final def getContext(): AbstractActor.ActorContext = context.asInstanceOf[AbstractActor.ActorContext]
```

Comment:
Do you see any good reason why those accessors should not be final? Does it make sense to override them?